src/libs/conduit/conduit_node_iterator.cpp

```diff
@@ -135,6 +137,24 @@ NodeIterator::index() const
 }
 
 //---------------------------------------------------------------------------//
+std::string
+NodeIterator::id() const
```

Comment:
@cyrush: I ran a test case just now to determine the exact behavior; attempting to call `name()` on a list iterator causes the following error:
```
{
  "file": /path/to/conduit/src/libs/conduit/conduit_schema.cpp",
  "line": 1191,
  "message": "<Schema::object_hierarchy[OBJECT_ID]>: Schema is not OBJECT_ID"
}
```
It looks like the `object_hierarchy` variable used by `name()` is only valid for objects.

If the `name()` function is supposed to return a valid string for list iterators, then I can just replace the `name()` implementation with the new `id()` function's contents.  If not, then I'm content keeping `id()` or renaming it to something more appropriate.